File: binance4j-mining/src/main/java/com/binance4j/mining/dto/WorkerStatus.java
package com.binance4j.mining.dto;

import io.swagger.annotations.ApiModelProperty;

/** Worker status. */
public enum WorkerStatus {
	/** All. */
	ALL("0"), @ApiModelProperty("")
	/** Valid. */
	VALID("1"), @ApiModelProperty("")
	/** Invalid. */
	INVALID("2"), @ApiModelProperty("")
	/** Failure. */
	FAILURE("3");

	String value;

	/**
	 * @param value
	 */
	private WorkerStatus(String value) {
		this.value = value;
	}

	@Override
	public String toString() {
		return value;
	}
}

File: connectors/src/main/kotlin/com/binance4j/mining/dto/WorkerStatus.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package com.binance4j.mining.dto

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Worker status.
 *
 * @property value
 */
enum class WorkerStatus(@JsonProperty("value")
	var value: String) {
	/** All. */
	ALL("0"),
	
	/** Valid. */
	VALID("1"),
	
	/**  Invalid. */
	INVALID("2"),
	
	/** Failure. */
	FAILURE("3");
	
	override fun toString(): String = value
}